custom/enikshay/ucr/views.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import absolute_import
+from __future__ import division
+
+from datetime import datetime
+
+from django.db.models import Min
+
+from corehq.apps.userreports.models import AsyncIndicator
+from corehq.apps.userreports.reports.view import CustomConfigurableReport
+
+
+class MonitoredReport(CustomConfigurableReport):
+    """For reports backed by an async datasource, shows an indication of how far
+    behind the report might be, in increments of 12 hours.
+
+    """
+
+    template_name = 'enikshay/ucr/monitored_report.html'
+
+    @property
+    def page_context(self):
+        context = super(MonitoredReport, self).page_context
+        context['hours_behind'] = self.hours_behind()
+        return context
+
+    def hours_behind(self):
+        """returns the number of hours behind this report is, to the nearest 12 hour bucket.
+        """
+        now = datetime.utcnow()
+
+        oldest_indicator = (
+            AsyncIndicator.objects
+            .filter(indicator_config_ids__contains=[self.spec.config_id])
+            .aggregate(Min('date_created'))
```

Comment:
I think it would be more simple to just order by date_created (that's actually an implicit ordering for this model) and take the first result.

Reply:
https://github.com/dimagi/commcare-hq/pull/19117/commits/076aa11e353440b0c61a763c4b1bb2e4b57b9a30